### PR TITLE
Use the New Endpoint `bulk/congresstrading?version=v2`

### PR DESCRIPTION
**Pull request by AlexCatarino**
- Improve Data Source Classes 
  - Adds new available properties: Party, District, and State.
  - Fixes Transaction Direction Json converter to handle cases such as "Sale (Full)", since it was converted to `OrderDirection.Hold`.

- Updates Data Downloader 
  - Use the new endpoint `bulk/congresstrading?version=v2`.
  - Handle asset type to filter out non-stock
  - Handle trade size range that feeds the Amount and MaximumAmount properties.

- Adds and Updates Unit Tests

### CODE REVIEW

File: DataProcessing/QuiverCongressDataDownloader.cs [hunk 1]
```diff
@@ -102,15 +101,16 @@ public bool Run()
...
+                    .DeserializeObject<List<RawQuiverCongressDataPoint>>(rawCongressData, _jsonSerializerSettings)!
                     .Where(x =>
                     {
                         // We don't have enough information to disambiguate whether this transaction,
                         // known as an "Exchange", is the acquisition or dumping of an asset.
                         // Also, ReportDate might be null, but we use it for setting the EndTime
                         // of the QuiverCongress type. So if it doesn't exist, we don't know
                         // when the data was made available to us.
-                        if (x.Transaction == OrderDirection.Hold || x.ReportDate == null || x.ReportDate > today)
+                        var isStock = !string.IsNullOrWhiteSpace(x.TickerType) || x.TickerType != "Stock" || x.TickerType != "ST";
```
Comment (jhonabreul) [inline DataProcessing/QuiverCongressDataDownloader.cs]: Should this be ` var isStock = !string.IsNullOrWhiteSpace(x.TickerType) && (x.TickerType == "Stock" || x.TickerType == "ST")`?

File: DataProcessing/QuiverCongressDataDownloader.cs [hunk 2]
```diff
@@ -270,7 +261,7 @@ private async Task<string> HttpRequester(string url)
 
             throw new Exception($"Request for {baseUrl}{url} failed with no more retries remaining (retry {MaxRetries}/{MaxRetries})");
         }
-
+
```
Comment (jhonabreul) [inline DataProcessing/QuiverCongressDataDownloader.cs]: Spaces

File: TransactionDirectionJsonConverter.cs
Comment (jhonabreul) [inline TransactionDirectionJsonConverter.cs:1]: Is this change required?

File: DataProcessing/QuiverCongressDataDownloader.cs [hunk 3]
```diff
@@ -217,49 +211,46 @@ public bool Run()
         /// <exception cref="Exception">Failed to get data after exceeding retries</exception>
         private async Task<string> HttpRequester(string url)
         {
-            url = Uri.EscapeDataString(url);
```
Comment (jhonabreul) [inline DataProcessing/QuiverCongressDataDownloader.cs:220]: No need for scaping anymore?

**Review by jhonabreul — COMMENTED** [review]
Nice! Leaving a new minor comments

File: DataProcessing/QuiverCongressDataDownloader.cs (hunk 1 above)
Comment (jhonabreul) [reply DataProcessing/QuiverCongressDataDownloader.cs]: Or if filtering out non-stock, should be `var isNotStock = string.IsNullOrWhiteSpace(x.TickerType) || (x.TickerType != "Stock" && x.TickerType != "ST")` and then in the if below should be `isNotStock` instead of `!isStock`

File: TransactionDirectionJsonConverter.cs
Comment (AlexCatarino) [reply TransactionDirectionJsonConverter.cs:1]: Yes. In their dataset, we have "Sell (Full)", so we need to use `Startswith` to handle "Sell (Full)" and "Sell".

File: DataProcessing/QuiverCongressDataDownloader.cs (hunk 3 above)
Comment (AlexCatarino) [reply DataProcessing/QuiverCongressDataDownloader.cs:220]: The new url has `?version=v2`, which we should not escape, or the rul will be invalid down the line.

File: DataProcessing/QuiverCongressDataDownloader.cs (hunk 1 above)
Comment (AlexCatarino) [reply DataProcessing/QuiverCongressDataDownloader.cs]: It's a stock if x.TickerType is null or whitespace (this is odd), and if it's not null or whitespace, then it must be "Stock" or "ST", so I think this is the easiest to read. `isNotStock` would be:

```csharp
var isNotStock = !string.IsNullOrWhiteSpace(x.TickerType) && !(x.TickerType == "Stock" || x.TickerType == "ST")
```
